packages/flutter/test/widgets/focus_manager_test.dart

```diff
@@ -745,4 +745,42 @@ void main() {
     await tester.pump();
     expect(parent1.focusedChild, equals(child2));
   });
+  testWidgets('Focus changes notify listeners.', (WidgetTester tester) async {
```

Comment:
Can you add tests for these two cases which are known to be problematic in the other notification scheme?

- Focus changed because the node with current focus calls unfocus()
- Focus changed because the tree was rebuilt and dropped the node with current focus

Reply:
OK, I added both of those.